### DISCUSSION TITLE
How to replace with redirect on an action

### DISCUSSION THREAD

**Discussion by davbrito** [remix-run/react-router#10672] [Q&A] [answered]
With the useNavigate API you can set the "replace" option to replace the entry on the history stack. On the other hand with an action on the router, when using a redirect response you don't have a way to do that.

I have a login route that makes the authentication on the action. So I want to redirect to a page when logged in, but replacing the login entry on the history.

**Comment by brophdawg11** [comment, marked as answer]
If you're using `RouterProvider` and `action` you can control this via [`<Form replace>`](https://reactrouter.com/en/main/components/form#replace) and the prop should persist through the redirect.
